File: web/src/api/nginx_node.js
import service from '@/utils/request'

// @Tags NginxNode
// @Summary 创建NginxNode
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.NginxNode true "创建NginxNode"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"获取成功"}"
// @Router /Node/createNginxNode [post]
export const createNginxNode = (data) => {
     return service({
         url: "/Node/createNginxNode",
         method: 'post',
         data
     })
 }


// @Tags NginxNode
// @Summary 删除NginxNode
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.NginxNode true "删除NginxNode"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"删除成功"}"
// @Router /Node/deleteNginxNode [delete]
 export const deleteNginxNode = (data) => {
     return service({
         url: "/Node/deleteNginxNode",
         method: 'delete',
         data
     })
 }

// @Tags NginxNode
// @Summary 删除NginxNode
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body request.IdsReq true "批量删除NginxNode"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"删除成功"}"
// @Router /Node/deleteNginxNode [delete]
 export const deleteNginxNodeByIds = (data) => {
     return service({
         url: "/Node/deleteNginxNodeByIds",
         method: 'delete',
         data
     })
 }

// @Tags NginxNode
// @Summary 更新NginxNode
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.NginxNode true "更新NginxNode"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"更新成功"}"
// @Router /Node/updateNginxNode [put]
 export const updateNginxNode = (data) => {
     return service({
         url: "/Node/updateNginxNode",
         method: 'put',
         data
     })
 }


// @Tags NginxNode
// @Summary 用id查询NginxNode
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.NginxNode true "用id查询NginxNode"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"查询成功"}"
// @Router /Node/findNginxNode [get]
 export const findNginxNode = (params) => {
     return service({
         url: "/Node/findNginxNode",
         method: 'get',
         params
     })
 }


// @Tags NginxNode
// @Summary 分页获取NginxNode列表
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body request.PageInfo true "分页获取NginxNode列表"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"获取成功"}"
// @Router /Node/getNginxNodeList [get]
 export const getNginxNodeList = (params) => {
     return service({
         url: "/Node/getNginxNodeList",
         method: 'get',
         params
     })
 }

File: server/router/nginx_node.go
package router

import (
	"nginx-web/api/v1"
	"nginx-web/middleware"
	"github.com/gin-gonic/gin"
)

func InitNginxNodeRouter(Router *gin.RouterGroup) {
	NginxNodeRouter := Router.Group("Node").Use(middleware.JWTAuth()).Use(middleware.CasbinHandler()).Use(middleware.OperationRecord())
	{
		NginxNodeRouter.POST("createNginxNode", v1.CreateNginxNode)   // 新建NginxNode
		NginxNodeRouter.DELETE("deleteNginxNode", v1.DeleteNginxNode) // 删除NginxNode
		NginxNodeRouter.DELETE("deleteNginxNodeByIds", v1.DeleteNginxNodeByIds) // 批量删除NginxNode
		NginxNodeRouter.PUT("updateNginxNode", v1.UpdateNginxNode)    // 更新NginxNode
		NginxNodeRouter.GET("findNginxNode", v1.FindNginxNode)        // 根据ID获取NginxNode
		NginxNodeRouter.GET("getNginxNodeList", v1.GetNginxNodeByClusterId)  // 获取NginxNode列表
	}
}
